pairing.py

```diff
@@ -3,66 +3,48 @@
+random = np.random.default_rng()
 
-class CollisionCounter:
+class Participant:
 
-    def __init__(self, num_people, gamma=1/np.e):
-        self.num_people = num_people
-        self.collission_counter = np.zeros(num_people)
-        self.log_gamma = np.log(gamma)
+    def __init__(self, num_people):
+        self.meetings_counter = np.zeros(num_people)
 
-    def increment_hits(self, j):
-        self.collission_counter[j] += 1
+    def increment_hits(self, *other_participants):
+        for other in other_participants:
+            self.meetings_counter[other] += 1
     
-    def sample_match(self, feasible):
-        feas = self._log_gamma*self._collission_counter[feasible]
+    def sample_match(self, indices, gamma):
+        feas = gamma*self.meetings_counter[indices]
         probabilities = softmax(feas)
-        match = choices(feasible, probabilities, k=1).pop()
+        match = random.choice(indices, p=probabilities)
         return match
 
-def generate_pairs(meeting_counts, num_people):
-    feasible = list(range(num_people))
+def generate_pairs(participants, gamma):
+    unmatched = list(range(len(participants)))
     pairs = []
-    while len(feasible) != 0:
-        index = np.random.choice(feasible)
-        feasible.remove(index)
-        cc = meeting_counts[index]
-        match = cc.sample_match(feasible)
-        feasible.remove(match)
-        pairs.append((index, match))
-    return pairs
-
-def generate_circuit(meeting_counts, num_people):
-    feasible = list(range(num_people))
-    pairs = []
-    start = np.random.choice(feasible)
-    index = start
-    while len(feasible) > 1:
-        feasible.remove(index)
-        cc = meeting_counts[index]
-        match = cc.sample_match(feasible)
-        pairs.append((index, match))
-        index = match
-    pairs.append((index,start))
+    while len(unmatched)>1:
+        index = random.choice(unmatched)
+        unmatched.remove(index)
+        participant = participants[index]
+        match = participant.sample_match(unmatched, gamma)
+        unmatched.remove(match)
+        pairs.append([index, match])
+    if len(unmatched) == 1:
+        pairs[0].append(unmatched[0])
     return pairs
 
 # sampling first person based on stuff
-def main(num_people, num_iters, gamma, circuit):
-    meeting_counts = [CollisionCounter(num_people, gamma=gamma) for k in range(num_people)]
-    for i in range(num_iters):
-        if circuit:
-            pairs = generate_circuit(meeting_counts, num_people)
-        else:
-            pairs = generate_pairs(meeting_counts, num_people)
-        for i,j in pairs:
-            meeting_counts[i].increment_hits(j)
-            meeting_counts[j].increment_hits(i)
-    print("average hit/person:",
-            (2 if circuit else 1)*num_iters/(num_people-1))
-    for i,q in enumerate(meeting_counts):
-        cc = q.collision_counter()
-        print("person {}:".format(i), cc)
+def main(num_people, num_iters, gamma):
+    gamma = np.log(gamma)
```

Comment:
nit: if we're doing this here can we change the `gamma` parameter in the method `main(...)` to something else? I feel like this line is super confusing without context